### PR TITLE
Chore: docs v1.0.0

### PR DESCRIPTION
**Pull request by r4mmer**

### CODE REVIEW

File: doc/TRANSACTION.md [hunk 1]
```diff
@@ -1,56 +1,62 @@
...
-## Links
+Deterministic ECDSA ([RFC 6979](https://tools.ietf.org/html/rfc6979)) is used to sign transaction on the [SECP-256k1](https://www.secg.org/sec2-v2.pdf#subsubsection.2.4.1) curve.
+The SDK call used should be `cx_ecdsa_sign` of the message.
+The message is everything after the `change_info` (the `sighash_all`).
 
-- Bitcoin Transaction: https://en.bitcoin.it/wiki/Protocol_documentation#tx
+## Address format
 
-- Ethereum Transaction: https://ethereum.github.io/yellowpaper/paper.pdf#subsection.4.2
+Hathor addresses are generated by b58 encodeding the hash160 of the compressed pubkey.
```
Comment (pedroferreira1) [inline doc/TRANSACTION.md:61]: Typo 'encodeding'

File: src/constants.h [hunk 2]
```diff
@@ -34,8 +34,3 @@
  * Pubkey hash length
  */
 #define PUBKEY_HASH_LEN 20
-
-/**
- * Maximum number of outputs on a transaction.
- */
-#define MAX_NUM_TX_OUTPUTS 255
```
Comment (pedroferreira1) [inline src/constants.h:41]: Why did you remove this?

Comment (r4mmer) [reply src/constants.h:41]: Unused since we do not test this anymore. The limit is enforced when we read the length of outputs as a 1 byte unsigned integer.

File: doc/TRANSACTION.md (hunk 1 above)
Comment (r4mmer) [reply doc/TRANSACTION.md]: Fixed

**Review by pedroferreira1 — APPROVED** [review]

File: doc/TRANSACTION.md [hunk 3]
```diff
@@ -1,56 +1,62 @@
...
-## Structure
+| Field | Size | Description |
+| --- | --- | --- |
+| `has_change_output` | 1bit | Indicates existence of a change output.<br>(mask `0x80`)<br>If false, the whole first byte is 0. |
+| `change bip32 path length` | 4bits | if `has_change_output=true` this will be the last 4 bits of the first byte<br>(mask `0x0F`) |
+| `change output index` | 1 byte | the index of the change output |
+| `change bip32 path` | var (`4*bip32_path_length`) | an array of 32bit unsigned integers of the path indexes |
 
-### Transaction
+Obs: bip32 paths have a hard depth limitation of 5, so the max value of the change info is 22 bytes
```
Comment (msbrogli) [inline doc/TRANSACTION.md:31]: Do we enforce this limitation in our code?

**Review by msbrogli — APPROVED** [review]

File: doc/TRANSACTION.md (hunk 3 above)
Comment (r4mmer) [reply doc/TRANSACTION.md:31]: Yes, on `src/common/bip32.c` we test the length with `MAX_BIP32_PATH` defined on `src/common/bip32.h` as 5

Comment (r4mmer) [reply doc/TRANSACTION.md:31]: We can allow longer paths (length up to 10 is the usual), but since we don't use derivations past this I think we shouldn't.
